android/src/main/kotlin/com/smileidentity/flutter/SmileIDSmartSelfieCaptureView.kt

```diff
@@ -56,22 +86,118 @@ internal class SmileIDSmartSelfieCaptureView private constructor(
+                                        showConfirmationDialog,
+                                        uiState,
+                                        viewModel,
+                                    )
+                                else -> RenderSelfieCaptureScreen(
+                                    userId,
+                                    jobId,
+                                    allowAgentMode,
+                                    viewModel,
+                                )
+                            }
                         }
                     },
                 )
             }
         }
     }
 
+    @Composable
+    private fun RenderSelfieCaptureScreen(
+        userId: String,
+        jobId: String,
+        allowAgentMode: Boolean,
+        viewModel: SelfieViewModel,
+    ) {
+        Box(
+            modifier = Modifier
+                .background(color = Color.White)
+                .windowInsetsPadding(WindowInsets.statusBars)
+                .consumeWindowInsets(WindowInsets.statusBars)
+                .fillMaxSize(),
+        ) {
+            SelfieCaptureScreen(
+                userId = userId,
+                jobId = jobId,
+                allowAgentMode = allowAgentMode,
+                allowNewEnroll = true,
+                skipApiSubmission = true,
+                viewModel = viewModel,
+            )
+        }
+    }
+
+    @Composable
+    private fun HandleSelfieConfirmation(
+        showConfirmation: Boolean,
+        uiState: SelfieUiState,
+        viewModel: SelfieViewModel,
+    ) {
+        if (showConfirmation) {
+            ImageCaptureConfirmationDialog(
+                titleText = stringResource(R.string.si_smart_selfie_confirmation_dialog_title),
+                subtitleText = stringResource(
+                    R.string.si_smart_selfie_confirmation_dialog_subtitle,
+                ),
+                painter = BitmapPainter(
+                    BitmapFactory
+                        .decodeFile(uiState.selfieToConfirm!!.absolutePath)
+                        .asImageBitmap(),
+                ),
+                confirmButtonText = stringResource(
+                    R.string.si_smart_selfie_confirmation_dialog_confirm_button,
+                ),
+                onConfirm = { viewModel.submitJob() },
+                retakeButtonText = stringResource(
+                    R.string.si_smart_selfie_confirmation_dialog_retake_button,
+                ),
+                onRetake = viewModel::onSelfieRejected,
+                scaleFactor = 1.25f,
+            )
+        } else {
+            viewModel.submitJob()
+        }
+    }
+
+    @Composable
+    private fun HandleProcessingState(viewModel: SelfieViewModel) {
+        viewModel.onFinished { res ->
+            when (res) {
+                is SmileIDResult.Success -> handleResult(res)
+                is SmileIDResult.Error -> onError(res.throwable)
+            }
+        }
```

Comment:
```suggestion
        viewModel.onFinished { res -> handleResult(res) },
```
I think we can do it like this, given that the `handleResult` is also able to process the error case - https://github.com/smileidentity/flutter/blob/1c6bdd665355543afc620d86cd36432c694b5983/android/src/main/kotlin/com/smileidentity/flutter/SmileSelfieComposablePlatformView.kt#L46